spec/forms/idv/finance_form_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+describe Idv::FinanceForm do
+  subject { Idv::FinanceForm.new({}) }
+
+  describe '#submit' do
+    it 'adds ccn key to idv_params when valid' do
+      expect(subject.submit(finance_account: '12345678', finance_type: :ccn)).to eq true
+
+      expected_params = {
+        ccn: '12345678'
+      }
+
+      expect(subject.idv_params).to eq expected_params
+    end
+
+    it 'fails when missing all finance fields' do
+      expect(subject.submit(foo: 'bar')).to eq false
+    end
+  end
+end
```

Comment:
Add specs to test the various validations? You can use shoulda_matchers for the presence validations.